bin/main.rb

```diff
@@ -1 +1,80 @@
-puts "Hello World!"
\ No newline at end of file
+def welcome_message
+  puts "Welcome to Tic Tac Toe Game"
+end
+welcome_message
+
+def user_setup
+  print "Please Enter Player One name "
+  player1 = gets.chomp
+  print "Please Enter Player Second name "
+  player2 = gets.chomp
+
+  print "Choose #{player1} X or O: "
+  mark = gets.chomp
+  mark.upcase!
+  if mark == "X"
+      puts "X is assigned to #{player1} and O is assigned to #{player2}"
+  else
+      puts "O is assigned to #{player1} and X is assigned to #{player2}"
+  end
+
+end
+
+user_setup
+
+puts "Game Board"
+board = [" "," "," "," "," "," "," "," "," "]
+def display_board(board)
+  puts " #{board[0]} | #{board[1]} | #{board[2]} "
+  puts "-----------"
+  puts " #{board[3]} | #{board[4]} | #{board[5]} "
+  puts "-----------"
+  puts " #{board[6]} | #{board[7]} | #{board[8]} "
+end
+
+def empty_index_board(board, index)
+    if board[index] == " " || board[index] == ""
+        true
+    else
+        false    
+    end
+end
+
+def choose_num1(board)
+    
+    arr = []
+
+    print "Please User 1 Choose number from 1 to 9: "
+    condition = true
+    while condition
+        
+    number1 = gets.to_i
+    if number1 <= 9 && number1 > 0 && empty_index_board(board, number1) == true 
+        condition = false
+        board[number1-1] = "X"
+        display_board(board)
+    else 
+        print "Enter valid number: "        
+    end
```

Comment:
#### There are 2 main fundamentals issue on this code: 
- [ ] notice how player 1 is allowed to overwrite the player 2 input value
- [ ] notice how player 1 will never be able to play the number 9 since you used the <= operator